src/theory/arith/arith_poly_norm.cpp

```diff
@@ -531,6 +533,11 @@ Node PolyNorm::getPolyNorm(Node a)
     PolyNorm pa = arith::PolyNorm::mkPolyNorm(a);
     an = pa.toNode(a.getType());
     a.setAttribute(apna, an);
+    // as an optimization, assume idempotent
+    if (a!=an)
+    {
+      an.setAttribute(apna, an);
+    }
```

Comment:
Format! Do we need the check? Is it not possible to set the attribute twice?

Reply:
I'd expect the check to be slightly faster (pointer comparison faster than map insert), where this method will generate a lot of a == an.

Reply:
OK. Please format!